server/src/main/resources/rules/rules.js

```diff
@@ -561,12 +561,13 @@ var FactValueCalculator = {
         },
 
         guest_limit: function (prodAttr, consumer) {
-            if (consumer.guestIds === null) {
+            var context = JSON.parse(json_context);
+            if (context.guestIds === null) {
                 return 0;
             }
             var activeGuestCount = 0;
-            for (var guestIdx = 0; guestIdx < consumer.guestIds.length; guestIdx++) {
-                var guest = consumer.guestIds[guestIdx];
+            for (var guestIdx = 0; guestIdx < context.guestIds.length; guestIdx++) {
```

Comment:
This is fine here, but going forward when we work with Javascript arrays, we should avoid accessing .length as part of a loop like this. With JS there is a bunch of work traversing objects to find the source of a given property, which is done on every access. As such, it's more performant to write loops like this to either traverse is descending order or store the length in a variable right before the loop:

```
for (var guestIdx = context.guestIds.length -1; guestIdx >= 0; --guestIdx) {

...

var guestIdLength = context.guestIds.length;
for (var guestIdx = 0; guestIdx < guestIdsLength; guestIdx++) {
```